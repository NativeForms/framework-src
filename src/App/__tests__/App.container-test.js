import React from 'react';
import Immutable from 'immutable';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';

// app
import AppContainer from '../App.container';
import initialState from '../Navigation/Navigation.state';

const mockStore = configureMockStore([]);

test('AppContainer should render correctly', () => {
  const wrapper = shallow(<AppContainer />, {
    context: { store: mockStore(Immutable.Map({ nav: initialState })) }
  });
  expect(wrapper.dive()).toMatchSnapshot();
});
