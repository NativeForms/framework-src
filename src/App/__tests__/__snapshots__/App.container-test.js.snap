// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppContainer should render correctly 1`] = `
<Styled(Container)>
  <NavigationContainer
    navigation={
      Object {
        "dispatch": [Function],
        "goBack": [Function],
        "navigate": [Function],
        "setParams": [Function],
        "state": Object {
          "index": 0,
          "routes": Array [
            Object {
              "index": 0,
              "key": "DrawerClose",
              "routeName": "DrawerClose",
              "routes": Array [
                Object {
                  "index": 2,
                  "key": "Home",
                  "routeName": "Home",
                  "routes": Array [
                    Object {
                      "key": "Init-id-42-0",
                      "routeName": "Home",
                    },
                    Object {
                      "key": "id-42-5",
                      "params": undefined,
                      "routeName": "Home",
                    },
                    Object {
                      "key": "id-42-6",
                      "params": undefined,
                      "routeName": "Home",
                    },
                  ],
                },
                Object {
                  "index": 0,
                  "key": "TextInput",
                  "routeName": "TextInput",
                  "routes": Array [
                    Object {
                      "key": "Init-id-42-1",
                      "routeName": "TextInput",
                    },
                  ],
                },
                Object {
                  "index": 0,
                  "key": "Button",
                  "routeName": "Button",
                  "routes": Array [
                    Object {
                      "key": "Init-id-42-2",
                      "routeName": "Button",
                    },
                  ],
                },
                Object {
                  "index": 0,
                  "key": "Link",
                  "routeName": "Link",
                  "routes": Array [
                    Object {
                      "key": "Init-id-42-3",
                      "routeName": "Link",
                    },
                  ],
                },
                Object {
                  "index": 0,
                  "key": "Login",
                  "routeName": "Login",
                  "routes": Array [
                    Object {
                      "key": "Init-id-42-4",
                      "routeName": "Login",
                    },
                  ],
                },
              ],
            },
            Object {
              "key": "DrawerOpen",
              "routeName": "DrawerOpen",
            },
          ],
        },
      }
    }
  />
</Styled(Container)>
`;
